packages/experimental/Link/src/useLink.ts

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { useAsPressable, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { LinkProps, LinkState } from './Link.types';
+import { memoize } from '@fluentui-react-native/framework';
+import { AccessibilityState, Linking, Platform } from 'react-native';
+
+export const useLink = (props: LinkProps): LinkState => {
+  // attach the pressable state handlers
+  const defaultComponentRef = React.useRef(null);
+  const {
+    accessibilityRole,
+    onPress,
+    onAccessibilityTap,
+    tooltip,
+    url,
+    accessibilityState,
+    componentRef = defaultComponentRef,
+    disabled,
+    enableFocusRing,
+    focusable,
+    ...rest
+  } = props;
+  const isDisabled = !!disabled;
+
+  const [visitedState, setVisitedState] = React.useState({ visited: false });
+  const linkOnPress = React.useCallback(
+    (e) => {
+      setVisitedState({ visited: true });
+      if (url) {
+        Linking.openURL(url as string);
+      } else if (onPress) {
+        onPress(e);
+      }
+      e.stopPropagation();
+    },
+    [setVisitedState, url, onPress],
+  );
+
+  // GH #1336: Set focusRef to null if link is disabled to prevent getting keyboard focus.
+  const focusRef = isDisabled && !focusable ? null : componentRef;
+  const onPressWithFocus = useOnPressWithFocus(focusRef, linkOnPress);
+  const pressable = useAsPressable({ ...rest, disabled: isDisabled, onPress: onPressWithFocus });
+  const onKeyUpProps = useKeyProps(linkOnPress, ' ', 'Enter');
+
+  const newState = {
+    ...pressable.state,
+    ...visitedState,
+  };
+  const onAccTap = React.useCallback(
+    (e?) => {
+      onAccessibilityTap ? onAccessibilityTap() : linkOnPress(e);
+    },
+    [linkOnPress, onAccessibilityTap],
+  );
+
+  const linkTooltip = tooltip ?? url ?? undefined;
+
+  return {
+    props: {
+      ...rest,
+      ...onKeyUpProps,
+      ...pressable.props, // allow user key events to override those set by us
+      accessible: true,
```

Comment:
Might want to make it so that the client can change this if they want to - not that it'd happen frequently